mesonbuild/compilers/mixins/gnu.py

```diff
@@ -469,16 +469,16 @@ def _split_fetch_real_dirs(self, pathstr: str) -> T.List[str]:
             # paths under /lib would be considered not a "system path",
             # which is wrong and breaks things. Store everything, just to be sure.
             pobj = pathlib.Path(p)
-            unresolved = pobj.as_posix()
             if pobj.exists():
-                if unresolved not in result:
-                    result.append(unresolved)
                 try:
-                    resolved = pathlib.Path(p).resolve().as_posix()
+                    resolved = pobj.resolve(True).as_posix()
```

Comment:
I think this won't work on Python 3.5 as it won't be expecting the `strict` argument to be present at all.  Not sure how to solve that except by bumping the minimum version or introspecting somehow.

Reply:
> I think this won't work on Python 3.5 as it won't be expecting the `strict` argument to be present at all. Not sure how to solve that except by bumping the minimum version or introspecting somehow.

meson now requires Python 3.7 minimum version